src/resources/jupyter/notebook.py

```diff
@@ -159,9 +159,16 @@ def notebook_execute(options, status):
       # progress
       progress = (not quiet) and cell.cell_type == 'code' and index > 0
       if progress:
-         status("  Cell {0}/{1}...".format(
-            current_code_cell- 1, total_code_cells - 1
-         ))
+        # get label
+        label = ''
+        source_lines = cell.source.split('\n')
+        label_line = next((line for line in source_lines if line.startswith('#| label:')), None)
+        if label_line:
+            label = label_line.replace('#| label:', '').strip()
+
+        status("  Cell '{0}' {1}/{2}...".format(
+           label, current_code_cell- 1, total_code_cells - 1
+        ))
```

Comment:
It should be more reliable to get the `id` straight from the cell.
Notebooks, will always have an identifier, but the `id` is replaced by `label` if any.
This would avoid to re-do the parsing of parameters, etc.

> Notebook cell identifier. Note that if there is no cell `id` then `label` will be used as the cell `id` if it is present. See https://jupyter.org/enhancement-proposals/62-cell-id/cell-id.html for additional details on cell ids.

```suggestion
         status("  Cell '{0}' {1}/{2}...".format(
            cell.id,
            current_code_cell - 1,
            total_code_cells - 1,
         )
```

Reply:
The above suggested code produces the following output:

<img width="774" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/d29e84c3-d82d-464b-a807-4cbdd7674f39">

Reply:
I think that cells with automatically generated IDs shouldn't show the label. I also think that we should measure for the max width of a cell label and pad the other output so all progress messages have the same width (i.e. the `Done` lines up across lines).

Reply:
I agree with the suggestion to standardise the messages.

A 80 width string filled with spaces and "Done" at the end would do?

Also other messages would need to be "width standardised".

Reply:
Dear @mcanouil @jjallaire,

Thank you for your feedback!

In the updated PR, I changed the output format and improved output alignment.

After reviewing the codebase and documentation, it seems that label information cannot be directly obtained as it has not been parsed before. Therefore, I've optimized my code using regex to extract label sets and store all cell labels from the current file in a list. This will help us during output and ensure alignment by obtaining the maximum label length.

Sample output:
```
  Cell 1/3: 'fig-polar-1'........Done
  Cell 2/3: 'fig-polar-test-2'...Done
  Cell 3/3: ''...................Done
```

Reply:
It's why, in my opinion, `cell.id` is the best, as the more robust, way to get the label as parsed earlier already by Quarto.
The trade off is that, you get a "label" for all cells.

PS: I am not maintainer of Quarto code base, only a contributor.

Reply:
Thank you for your explanation! 

And I now have a more clear understanding of `cell.id`. Upon further research into it, I discovered that when a label is present in the cell, the `cell.id` represents the label, whereas it represents the id itself when no label is available. I must acknowledge that I had not thoroughly investigated `cell.id` prior to this, and I sincerely apologize for any oversight on my part. I truly appreciate your patience and elucidation on this matter!

Here is the updated sample output in the latest PR:
```
  Cell 1/3: 'fig-polar-1'......Done
  Cell 2/3: 'fig-polar-1024'...Done
  Cell 3/3: 'a1886270'.........Done
```

Reply:
I still think we ought to not emit `a1886270` and other automatically-generated IDs. If at that point in the code it's too late to know which ID is automatically generated and which isn't, then we need to find a way to carry this information into the right part of the code.

Reply:
Thank you for bringing this to my attention! Yes, I agree with the suggestion that perhaps there needs to be a flag for whether the ID is automatically generated or not. The reason is, according to the documentation:

define: `cell_id` is the id in the output.

```
if "#| id":
   cell_id = cell.id
elif "#| label":
   cell_id = cell.id
else:
   cell_id = <auto-generated-id>
```

Since `cell.id` is automatically generated when neither `"#| id"` nor `"#| label"` exists, the above logic is equivalent to:

```
if is_auto_generated(cell.id) == False:
   cell_id = cell.id
else:
   cell_id = <auto-generated-id>
```

Therefore, once we have a flag to know if the id is auto-generated, we can set cell_id to `""`:
```
if is_auto_generated(cell.id) == False:
   cell_id = cell.id
else:
   cell_id = ""
```

We may need to modify this part of the code as well as some code that has called this function:
[quarto-dev/quarto-cli@58a9462](https://github.com/quarto-dev/quarto-cli/blob/main/src/core/jupyter/jupyter.ts#L583-L600)

Could you kindly provide some guidance on how the relevant code should be modified? Thank you!

ps. please ignore the latest commit (`3cea7dc`) in this branch.

Reply:
We may actually want to marry this with some work that @dragonstyle has been doing on notebook embedding. There, we want people to be able to embed notebook cells by a unique id. We have an idiom there where we first take the `label`, then use the first cell tag (if any). Perhaps we need a few field that represents this embeddable identifier?

Reply:
Thank you! While I was examining the code related to notebook embedding, I found the `nb_cell_yaml_options()` function. This led me to submit a commit ([cc1c803](https://github.com/quarto-dev/quarto-cli/pull/5772/commits/cc1c803467f794c6efb1b341eddec62b662b4b4f)) that might resolve the issue [[FR] Replace "Cell 1/N" by label (if provided) when using Jupyter #5752](https://github.com/quarto-dev/quarto-cli/issues/5752). Furthermore, I've noticed a potential problem with the `nb_cell_yaml_options()` function. Here's the summary:

**Summary of the Latest commit**

This commit introduces a feature where the render status will now display the label when a user has specified `"#| label"` for a "code" cell. When running the `"quarto render"` command, the output will match the following example:

```
  Cell 1/3: 'fig-polar-1'......Done
  Cell 2/3: 'fig-polar-1024'...Done
  Cell 3/3: ''.................Done
```

Notably, even if a user writes `"#| id"` in the "code" cell, the id will not be visible in the render status. 

This is because the return list from `nb_cell_yaml_options()` just includes cell attributes like `label`, `class`, `fig-cap`, `fig-subcap`, and so on, as specified in [cell-attributes.yml](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/schema/cell-attributes.yml) and [yaml-intelligence-resources.json](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/editor/tools/yaml/yaml-intelligence-resources.json), and does not include cell attributes like `"#| id"` `"#| tags"` and `"#| export"`. I'm uncertain if this is an issue that needs addressing or if there's a different approach to access these attributes. Any advice on this would be greatly appreciated!

**Additional Note**

I observed that in a recent PR ([Support export: in jupyter cells #4958](https://github.com/quarto-dev/quarto-cli/pull/4958)), we added support for `export`, but this attribute is currently absent from the official documentation [Code Cells: Jupyter](https://quarto.org/docs/reference/cells/cells-jupyter.html). I believe an update to the documentation to include this attribute might be beneficial.

Reply:
Then `export` syntax is a pre-processing directive used by nbdev so indeed not part of our schema or feature set. It (and the others ) can indeed be safely ignored here!